pkg/collectors/devices/ptp_devices_test.go

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package devices_test
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/clients"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/collectors/devices"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/testutils"
+)
+
+var testPod = &v1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:        "TestPod-8292",
+		Namespace:   "TestNamespace",
+		Annotations: map[string]string{},
+	},
+}
+
+var _ = Describe("NewContainerContext", func() {
+	var clientset *clients.Clientset
+	var response map[string][]byte
+	BeforeEach(func() {
+		clientset = testutils.GetMockedClientSet(testPod)
+		response = make(map[string][]byte)
+		responder := func(method string, url *url.URL) ([]byte, []byte, error) {
+			cmd := url.Query()["command"]
+			return response[strings.Join(cmd, " ")], []byte(""), nil
+		}
+		clients.NewSPDYExecutor = testutils.NewFakeNewSPDYExecutor(responder, nil)
+	})
+
+	When("called GetPTPDeviceInfo", func() {
+		It("should return a valid PTPDeviceInfo", func() {
+			vendor := "0x8086"
+			devID := "0x1593"
+			gnssDev := "gnss0"
+
+			response["ls /sys/class/net/aFakeInterface/device/gnss/"] = []byte("gnss0")
```

Comment:
Why the hard-coded `"gnss0"` here as well as on line 48? This should reference the `gnssDev` variable.